blog/nse_portfolio_sorts/index.qmd

```diff
@@ -0,0 +1,931 @@
+
+We have a grid with 69,120 specifications. For each of these specifications, we want to estimate a return differential. This is most easily achieved with a `pmap()` call. However, we want to parallelize the operation to leverage the multiple cores of our device. Hence, we have to use the package `furrr` and a `future_pmap()` instead. As a side note, I also have to increase my maximum memory for each worker, which I do with the call of `options()`. 
+
+```{r}
+#| eval: false
+library(furrr)
+
+options(future.globals.maxSize= 891289600)
+
+plan(multisession, workers = availableCores())
+```
+
+With the parallel environment set and ready to go, we map the arguments our function needs into the final function `execute_sorts()` from above. Then, we go and have some tea. And some lunch, breakfast, second breakfast, and so on. In short, it takes a while - depending on your device even more than a day. Each result requires roughly 13 seconds, but you must remember that you are computing 69,120 results. 
+
+```{r}
+#| eval: false
+data_premia <- setup_grid |>
+  mutate(premium_estimate = future_pmap(
+    .l = list(
+      sorting_variable, drop_smallNYSE_at, include_financials, include_utilities, drop_bookequity,
+      drop_earnings, drop_stock_age_at, drop_price_at, sv_lag, formation_time, n_portfolios_main,
+      sorting_method, n_portfolios_secondary, exchanges, value_weighted
+    ),
+    .f = ~ execute_sorts(
+      sorting_variable = ..1,
+      drop_smallNYSE_at = ..2,
+      include_financials = ..3,
+      include_utilities = ..4,
+      drop_bookequity = ..5,
+      drop_earnings = ..6,
+      drop_stock_age_at = ..7,
+      drop_price_at = ..8,
+      sv_lag = ..9,
+      formation_time = ..10,
+      n_portfolios_main = ..11,
+      sorting_method = ..12,
+      n_portfolios_secondary = ..13,
+      exchanges = ..14,
+      value_weighted = ..15
+    )
+  ))
+```
+
+At this point, we can also consider storing the data in our SQLite database for future reference. I skip this step today.
+
+```{r}
+#| include: false
+# From NSE database
+# data_premium_results <- dbReadTable(data_nse, "data_premium_results")
+# 
+# data_premia <- data_premium_results |> 
+#   filter(sorting_variable == "sv_ag") |> 
+#   select(sorting_variable, drop_smallNYSE_at, include_financials, include_utilities, drop_bookequity,
+#       drop_earnings, drop_stock_age_at, drop_price_at, sv_lag, formation_time, n_portfolios_main,
+#       sorting_method, n_portfolios_secondary, exchanges, value_weighted, premium_estimate = mean)
+# 
+# data_premia |> dbWriteTable(conn = data_tidy_nse, 
+#                name = "data_premia", 
+#                value = _,
+#                overwrite = TRUE)
+
+data_premia <- dbReadTable(data_tidy_nse, "data_premia")
+```
+
+# The premium distribution
+
+Given all the estimates for the premium, we can now take a look at their distribution with a call to `geom_density()` - of course, in the main Tidy Finance color scheme.
+
```

Comment:
"- of course, in the main Tidy Finance color scheme" streichen. Die Farben sind eh automatisch weil das pre_session.R script im _quarto.yml für das gesamte project referenziert ist, denke ich.

Reply:
Hat nicht geklappt, auch wenn ich das pre_render_script im Blogpost lade. Keine Ahnung warum, aber ich bin happy so wie es ist. 

@christophscheuch Soll ich den Satz trotzdem streichen?

Reply:
Done, comes with re-render commit.